test/unit/lib/vehicle.js

```diff
@@ -489,53 +472,112 @@ test('charge - no age', async function(t) {
   const response = await vehicle.charge();
   t.deepEqual(response.data, body);
   t.is(response.age, null);
-
 });
 
 test('vin', async function(t) {
-
   const body = {
     vin: '4JGBB8GB2AA537355',
   };
-  t.context.n = nocks.base()
+  t.context.n = nocks
+    .base()
     .get('/vin')
     .reply(200, body);
 
   const vin = await vehicle.vin();
   t.is(vin, body.vin);
-
 });
 
 test('lock', async function(t) {
-
-  t.context.nock = nocks.base()
+  t.context.nock = nocks
+    .base()
     .post('/security', {action: 'LOCK'})
     .reply(200, {status: 'success'});
 
   const response = await vehicle.lock();
 
-  t.deepEqual(
-    _.xor(_.keys(response), [
-      'status',
-    ]),
-    []
-  );
+  t.deepEqual(_.xor(_.keys(response), ['status']), []);
   t.is(response.status, 'success');
 });
 
 test('unlock', async function(t) {
-
-  t.context.nock = nocks.base()
+  t.context.nock = nocks
+    .base()
     .post('/security', {action: 'UNLOCK'})
     .reply(200, {status: 'success'});
 
   const response = await vehicle.unlock();
 
-  t.deepEqual(
-    _.xor(_.keys(response), [
-      'status',
-    ]),
-    []
-  );
+  t.deepEqual(_.xor(_.keys(response), ['status']), []);
   t.is(response.status, 'success');
 });
+
+test('batch', async function(t) {
+  vehicle.setUnitSystem('imperial');
+  const paths = ['/odometer', '/transmission/fluid', '/fuel', '/sunroof'];
+  const requestBody = {
+    headers: {
+      'sc-unit-system': 'imperial',
+    },
+    requests: [
+      {
+        path: '/odometer',
+      },
+      {
+        path: '/transmission/fluid',
+      },
+      {
+        path: '/fuel',
+      },
+      {
+        path: '/sunroof',
+      },
+    ],
+  };
+  const responseBody = {
+    responses: [
+      {
+        headers: {'sc-unit-system': 'imperial'},
+        path: '/odometer',
+        code: 200,
+        body: {
+          distance: 32768,
+        },
+      },
+      {
+        headers: {'sc-unit-system': 'imperial'},
+        path: '/transmission/fluid',
+        code: 200,
+        body: {
+          temperature: 98.2,
+          wear: 0.5,
+        },
+      },
+      {
+        headers: {'sc-unit-system': 'imperial'},
+        path: '/fuel',
+        code: 200,
+        body: {
+          range: 550.8499755859375,
+          percentRemaining: 0.9449999928474426,
+        },
+      },
+      {
+        headers: {'sc-unit-system': 'imperial'},
+        path: '/sunroof',
+        code: 501,
+        body: {
+          error: 'vehicle_not_capable_error',
+          message: 'Vehicle is not capable of performing request.',
+        },
+      },
+    ],
+  };
+
+  t.context.n = nocks
+    .base()
+    .post('/batch', requestBody)
+    .reply(200, responseBody);
+
+  const response = await vehicle.batch(paths);
```

Comment:
nit: Add an extra line after here.

We try to follow the AAA unit testing pattern (https://medium.com/@pjbgf/title-testing-code-ocd-and-the-aaa-pattern-df453975ab80), so ideally we separate each section that way its easier to read and review the tests!

Reply:
Fixed. I definitely agree that we should stick to AAA!